src/main/java/io/wispforest/owo/util/pond/MatrixStackManipulator.java

```diff
@@ -1,76 +1,70 @@
 package io.wispforest.owo.util.pond;
 
 import net.minecraft.client.gui.DrawContext;
+import net.minecraft.client.util.math.MatrixStack;
 import org.joml.Matrix4f;
 import org.joml.Quaternionf;
 
-public interface MatrixStackManipulator<C extends DrawContext> {
+/**
+ * Helper Interface implemented on top of the DrawContext through a Mixin allowing for easier matrix stack manipulation
+ */
+public interface MatrixStackManipulator<C extends MatrixStackManipulator<?>> {
 
-    default C translate(double x, double y, double z) {
-        C context = getContext();
+    static <C extends MatrixStackManipulator<?>> C of(DrawContext c){
+        return (C) c;
+    }
+
+    //----
 
-        context.getMatrices().translate(x, y, z);
+    default C translate(double x, double y, double z) {
+        getMatrices().translate(x, y, z);
```

Comment:
Please respect the style of the rest of this codebase and explicitly mention `this`
```suggestion
        this.getMatrices().translate(x, y, z);
```